File: RepoTransBench/source_projects/Java/awslabs_route53-infima/src/test/java/com/amazonaws/services/route53/infima/LatticePublicTest.java
package com.amazonaws.services.route53.infima;

import org.junit.jupiter.api.Test;

import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

class LatticePublicTest {

    @Test
    void testConstructionAndDimensionNamesWithDifferentDims() {
        List<String> dims = Arrays.asList("dimA", "dimB");
        Lattice<Integer> lattice = new Lattice<>(dims);
        assertEquals(dims, lattice.getDimensionNames());
    }

    @Test
    void testAddEndpointsForSectorAndGetEndpointsDifferent() {
        Lattice<String> lattice = new Lattice<>(Collections.singletonList("y"));
        List<String> coord = Collections.singletonList("custom");
        lattice.addEndpointsForSector(coord, Arrays.asList("P", "Q"));
        assertTrue(lattice.getEndpointsForSector(coord).contains("P"));
        assertTrue(lattice.getEndpointsForSector(coord).contains("Q"));
    }

    @Test
    void testSimulateFailureDifferentValues() {
        Lattice<String> lattice = new Lattice<>(Arrays.asList("country", "city"));
        List<String> coord1 = Arrays.asList("Spain", "Madrid");
        lattice.addEndpointsForSector(coord1, Arrays.asList("Server1", "Server2"));

        Lattice<String> failed = lattice.simulateFailure("country", "Spain");
        assertTrue(failed.getAllEndpoints().isEmpty() || failed.getEndpointsForSector(Arrays.asList("Spain", "Madrid")).isEmpty());
    }

    @Test
    void testGetDimensionValuesAndDimensionalityPublic() {
        Lattice<String> lattice = new Lattice<>(Arrays.asList("continent", "region"));
        lattice.addEndpointsForSector(Arrays.asList("Europe", "North"), Arrays.asList("HostA"));
        lattice.addEndpointsForSector(Arrays.asList("Europe", "South"), Arrays.asList("HostB"));
        Set<String> regions = new HashSet<>(lattice.getDimensionValues("region"));
        assertTrue(regions.contains("North"));
        assertTrue(regions.contains("South"));
        Map<String, Integer> dims = lattice.getDimensionality();
        assertEquals(2, dims.size());
    }

    @Test
    void testEqualsAndHashCodeDifferentData() {
        Lattice<String> l1 = new Lattice<>(Arrays.asList("R", "S"));
        Lattice<String> l2 = new Lattice<>(Arrays.asList("R", "S"));
        l1.addEndpointsForSector(Arrays.asList("foo", "bar"), Arrays.asList("baz"));
        l2.addEndpointsForSector(Arrays.asList("foo", "bar"), Arrays.asList("baz"));
        assertEquals(l1, l2);
        assertEquals(l1.hashCode(), l2.hashCode());
    }

    @Test
    void testToStringNotNullPublic() {
        Lattice<String> lattice = new Lattice<>(Collections.singletonList("E"));
        assertNotNull(lattice.toString());
    }

    @Test
    void testNoEndpointsPublic() {
        Lattice<String> lattice = new Lattice<>(Collections.singletonList("F"));
        assertTrue(lattice.getAllEndpoints().isEmpty());
    }
}

File: RepoTransBench/target_projects/Java/C#/awslabs_route53-infima/public_tests/LatticePublicTest.cs
using System;
using System.Collections.Generic;
using Xunit;

namespace Route53Infima.PublicTests
{
    public class LatticePublicTest
    {
        [Fact]
        public void TestConstructionAndDimensionNamesWithDifferentDims()
        {
            var dims = new List<string> { "dimA", "dimB" };
            var lattice = new Lattice<int>(dims);
            Assert.Equal(dims, lattice.GetDimensionNames());
        }

        [Fact]
        public void TestAddEndpointsForSectorAndGetEndpointsDifferent()
        {
            var lattice = new Lattice<string>(new List<string> { "y" });
            var coord = new List<string> { "custom" };
            lattice.AddEndpointsForSector(coord, new List<string> { "P", "Q" });
            var endpoints = lattice.GetEndpointsForSector(coord);
            Assert.Contains("P", endpoints);
            Assert.Contains("Q", endpoints);
        }

        [Fact]
        public void TestSimulateFailureDifferentValues()
        {
            var lattice = new Lattice<string>(new List<string> { "country", "city" });
            var coord = new List<string> { "Spain", "Madrid" };
            lattice.AddEndpointsForSector(coord, new List<string> { "Server1", "Server2" });

            var failed = lattice.SimulateFailure("country", "Spain");
            Assert.True(failed.GetAllEndpoints().Count == 0 ||
                        failed.GetEndpointsForSector(coord).Count == 0);
        }

        [Fact]
        public void TestGetDimensionValuesAndDimensionalityPublic()
        {
            var lattice = new Lattice<string>(new List<string> { "continent", "region" });
            lattice.AddEndpointsForSector(new List<string> { "Europe", "North" }, new List<string> { "HostA" });
            lattice.AddEndpointsForSector(new List<string> { "Europe", "South" }, new List<string> { "HostB" });

            var regions = new HashSet<string>(lattice.GetDimensionValues("region"));
            Assert.Contains("North", regions);
            Assert.Contains("South", regions);

            var dims = lattice.GetDimensionality();
            Assert.Equal(2, dims.Count);
        }

        [Fact]
        public void TestEqualsAndHashCodeDifferentData()
        {
            var l1 = new Lattice<string>(new List<string> { "R", "S" });
            var l2 = new Lattice<string>(new List<string> { "R", "S" });
            l1.AddEndpointsForSector(new List<string> { "foo", "bar" }, new List<string> { "baz" });
            l2.AddEndpointsForSector(new List<string> { "foo", "bar" }, new List<string> { "baz" });
            Assert.Equal(l1, l2);
            Assert.Equal(l1.GetHashCode(), l2.GetHashCode());
        }

        [Fact]
        public void TestToStringNotNullPublic()
        {
            var lattice = new Lattice<string>(new List<string> { "E" });
            Assert.NotNull(lattice.ToString());
        }

        [Fact]
        public void TestNoEndpointsPublic()
        {
            var lattice = new Lattice<string>(new List<string> { "F" });
            Assert.Empty(lattice.GetAllEndpoints());
        }
    }
}